java/src/main/native/src/RmmJni.cpp

```diff
@@ -26,12 +26,120 @@ using rmm::mr::device_memory_resource;
 
 namespace {
 
+// Alignment to which the RMM memory resource will round allocation sizes
+constexpr std::size_t RMM_ALLOC_SIZE_ALIGNMENT = 512;
```

Comment:
Technically resources are required to align only to 256B.

Reply:
I based this on CNMEM's behavior here: https://github.com/NVIDIA/cnmem/blob/master/src/cnmem.cpp#L48

Reply:
Yep, CNMEM does indeed use 512, just pointing out that other resources can and do use 256.